### PR TITLE
refactor: scale nimble and poise stamina modifier threshold

### PR DESCRIPTION
**Pull request by LordMidas**
Nimble is at 15% while Poise is at 35% of base fatigue including the effects of traits and permanent injuries.

### CODE REVIEW

**Comment by Uberbagel** [comment]
If not already, we should probably make the effect show on the character screen showing your currently allowed weight.

File: mod_reforged/hooks/config/strings.nut [hunk 1]
```diff
@@ -1556,8 +1556,7 @@ foreach (vanillaDesc in vanillaDescriptions)
 			Type = ::UPD.EffectType.Passive,
 			Description = [
 				"Damage to [Hitpoints|Concept.Hitpoints] is reduced by " + ::MSU.Text.colorPositive("30%") + " and to Armor by " + ::MSU.Text.colorPositive("20%") + ".",
-				"The bonus drops exponentially when wearing head and body armor with a total penalty to [Maximum Fatigue|Concept.MaximumFatigue] above 35.",
-				"Does not affect damage from mental attacks or status effects, but can help to avoid receiving them.",
+				"The bonus drops exponentially when wearing head and body armor with a total penalty to [Maximum Fatigue|Concept.MaximumFatigue] above 35% of your [Base|Concept.BaseAttribute] [Fatigue|Concept.Fatigue] including [traits|Concept.Trait] and [permanent injuries|Concept.InjuryPermanent].",				"Does not affect damage from mental attacks or status effects, but can help to avoid receiving them.",
```
Comment (Darxo) [inline mod_reforged/hooks/config/strings.nut:1559]: You are missinga newline between those two tooltip lines now

File: mod_reforged/hooks/skills/perks/perk_nimble.nut [hunk 2]
```diff
@@ -91,4 +98,14 @@
...
 	}
+
+	q.getStaminaModifierThreshold <- function()
+	{
+		local b = this.getContainer().getActor().getBaseProperties().getClone();
+		foreach (trait in this.getContainer().getSkillsByFunction(@(s) s.isType(::Const.SkillType.Trait) || s.isType(::Const.SkillType.PermanentInjury)))
+		{
+			trait.onUpdate(b);
+		}
+		return ::Math.max(0, 0.15 * b.Stamina * (b.StaminaMult >= 0 ? b.StaminaMult : 1.0 / b.StaminaMult));
```
Comment (Darxo) [inline mod_reforged/hooks/skills/perks/perk_nimble.nut:109]: the `0.15` should be a member variable for easier changing by future balancing commits or submods.

File: scripts/skills/perks/perk_rf_poise.nut [hunk 3]
```diff
@@ -102,4 +109,14 @@ this.perk_rf_poise <- ::inherit("scripts/skills/skill", {
...
 	}
+
+	q.getStaminaModifierThreshold <- function()
+	{
+		local b = this.getContainer().getActor().getBaseProperties().getClone();
+		foreach (trait in this.getContainer().getSkillsByFunction(@(s) s.isType(::Const.SkillType.Trait) || s.isType(::Const.SkillType.PermanentInjury)))
+		{
+			trait.onUpdate(b);
+		}
+		return ::Math.max(0, 0.35 * b.Stamina * (b.StaminaMult >= 0 ? b.StaminaMult : 1.0 / b.StaminaMult));
```
Comment (Darxo) [inline scripts/skills/perks/perk_rf_poise.nut:120]: the `0.35` should be a member variable for easier changing by future balancing commits or submods.

**Review by Darxo — CHANGES_REQUESTED** [review]

**Pull request merged**